source/plugins/ruby/lib/application_insights/channel/sender_base.rb

```diff
@@ -66,19 +66,19 @@ def send(data_to_send)
         request.body = compressed_data
         if @proxy.nil? || @proxy.empty?
           http = Net::HTTP.new uri.hostname, uri.port
-        else 
+        else
           http = Net::HTTP.new(uri.hostname, uri.port, @proxy[:addr], @proxy[:port], @proxy[:user], @proxy[:pass])
         end
-        if uri.scheme.downcase == 'https'
+        if uri.scheme.downcase == "https"
           http.use_ssl = true
-          http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+          http.verify_mode = OpenSSL::SSL::VERIFY_PEER
```

Comment:
This needs to validated in airgapped clouds

Reply:
> This needs to validated in airgapped clouds

>  this shouldnt impact on airgapped clouds since we do all other places. This is just server cert validation  to prevent MITM attacks which was not done AI ruby SDK and all other places we do it for example AI go sdk and 

https://github.com/microsoft/Docker-Provider/blob/ci_dev/source/plugins/ruby/KubernetesApiClient.rb#L50
https://github.com/microsoft/Docker-Provider/blob/ci_dev/build/windows/installer/scripts/rubyKeepCertificateAlive/omsagenthelper.rb#L94

Reply:
@ganga1980 - This is the trusted CA validation that happens when you change this to verify peer.. This has caused problems for ods and oms endpoints. Do you know which CA is used in AIrgapped clouds? This is generally different in these clouds and may not be in the trusted store and connections will fail.

Reply:
created a task to handle this. let me reach out Joby on the instructions to get access.